pkg/pillar/cmd/zedrouter/networkinstance.go

```diff
@@ -735,6 +748,31 @@ func doNetworkInstanceModify(ctx *zedrouterContext,
 	}
 }
 
+func checkNIphysicalPort(ctx *zedrouterContext, status *types.NetworkInstanceStatus) error {
+	// check the NI have the valid physical port binding to
+	label := status.Logicallabel
+	if label != "" && !strings.EqualFold(label, "uplink") &&
+		!strings.EqualFold(label, "freeuplink") {
+		var foundMatch bool
+		pub := ctx.subDeviceNetworkStatus
+		items := pub.GetAll()
+		for _, st := range items {
```

Comment:
No need to ask pubsub. Just use ctx.deviceNetworkStatus

Reply:
ok.